src/main/java/de/adesso/kicker/notification/NotificationService.java

```diff
@@ -1,56 +1,169 @@
+    private TournamentJoinRequestService tournamentJoinRequestService;
+    private MatchCreationRequestService matchCreationRequestService;
+    private MatchVerificationRequestService matchVerificationRequestService;
 
     @Autowired
-    public NotificationService(NotificationRepository notificationRepository) {
+    public NotificationService(NotificationRepository notificationRepository, UserService userService,
+            TeamJoinRequestService teamJoinRequestService, TournamentJoinRequestService tournamentJoinRequestService,
+            MatchCreationRequestService matchCreationRequestService,
+            MatchVerificationRequestService matchVerificationRequestService) {
+
         this.notificationRepository = notificationRepository;
+        this.userService = userService;
+        this.teamJoinRequestService = teamJoinRequestService;
+        this.tournamentJoinRequestService = tournamentJoinRequestService;
+        this.matchCreationRequestService = matchCreationRequestService;
+        this.matchCreationRequestService = matchCreationRequestService;
     }
 
-    // Gets a notification by id
     public Notification getNotificationById(long id) {
 
         return notificationRepository.findByNotificationId(id);
     }
 
-    // Get all notifications
     public List<Notification> getAllNotifications() {
 
         List<Notification> notifications = new ArrayList<>();
         notificationRepository.findAll().forEach(notifications::add);
         return notifications;
     }
 
-    // Get all notifications that were send by a specific user
     public List<Notification> getAllNotificationsBySender(User sender) {
 
         List<Notification> notifications = new ArrayList<>();
-        notificationRepository.findBySender(sender).forEach(notifications::add);
+        if (sender != null) {
+            notifications = getAllNotificationsBySender(sender.getUserId());
+        }
         return notifications;
     }
 
-    // Get all notifications that were received by a specific user
     public List<Notification> getAllNotificationsByReceiver(User receiver) {
 
+        List<Notification> notifications = new ArrayList<>();
+        if (receiver != null) {
+            notifications = getAllNotificationsByReceiver(receiver.getUserId());
+        }
+        return notifications;
+    }
+
+    public List<Notification> getAllNotificationsBySender(long senderId) {
+
+        User sender = userService.getUserById(senderId);
+        List<Notification> notifications = new ArrayList<>();
+        notificationRepository.findBySender(sender).forEach(notifications::add);
+        return notifications;
+    }
+
+    public List<Notification> getAllNotificationsByReceiver(long receiverId) {
+
+        User receiver = userService.getUserById(receiverId);
         List<Notification> notifications = new ArrayList<>();
         notificationRepository.findByReceiver(receiver).forEach(notifications::add);
```

Comment:
Can do a parametrized constructor call
```` java
return new ArrayList<>(notificationRepository.findByReceiver(receiver));
````
There are multiple places like this. Your IDE should highlight them.